src/integrationTests/options.test.ts

```diff
@@ -209,11 +219,332 @@ describe('useCookie option', () => {
+      expect(setTimeout).toBeCalledWith(expect.any(Function), 67890);
+    });
+  });
+
+  describe('usecookie changing from true to false', () => {
+    beforeEach(() => {
+      uid2.init({
+        callback: callback,
+        identity: identity,
+        baseUrl: baseUrl,
+        cookiePath: cookiePath,
+        refreshRetryPeriod: 12345,
+        useCookie: true,
+      });
+      uid2.init({
+        useCookie: false,
+      });
+    });
+    test('should change config from cookie to local storage', () => {
+      const storageConfig = getConfigStorage();
+      expect(storageConfig).toBeInstanceOf(Object);
+      expect(storageConfig).toHaveProperty('cookiePath');
+      const configCookie = getConfigCookie();
+      expect(configCookie).toBeNull();
+      expect(getUid2LocalStorage().advertising_token).toBe(identity.advertising_token);
+      expect(getUid2Cookie()).toBeNull();
+    });
+  });
+
+  describe('usecookie changing from false to true', () => {
+    beforeEach(() => {
+      uid2.init({
+        callback: callback,
+        identity: identity,
+        baseUrl: baseUrl,
+        refreshRetryPeriod: 12345,
+        useCookie: false,
+      });
+      uid2.init({
+        useCookie: true,
+        cookiePath: cookiePath,
+      });
+    });
+    test('should change config from cookie to local storage', () => {
+      const storageConfig = getConfigStorage();
+      expect(storageConfig).toBeNull();
+      const configCookie = getConfigCookie();
+      expect(configCookie).toBeInstanceOf(Object);
+      expect(configCookie).toHaveProperty('cookiePath');
+      expect(getUid2Cookie().advertising_token).toBe(identity.advertising_token);
+      expect(getUid2LocalStorage()).toBeNull();
+    });
+  });
+
+  describe('adding a callback when no callbacks exist before', () => {
+    beforeEach(() => {
+      uid2.init({
+        identity: identity,
+        baseUrl: baseUrl,
+        cookiePath: cookiePath,
+        refreshRetryPeriod: 12345,
+        useCookie: true,
+      });
+      uid2.init({
+        useCookie: false,
+        callback: callback,
+      });
+    });
+    test('should contain one init callback', () => {
+      const initCallbacks = uid2.getInitCallbacks();
+      expect(initCallbacks).toEqual([callback]);
+    });
+  });
+
+  describe('adding multiple callbacks that are different', () => {
+    beforeEach(() => {
+      uid2.init({
+        callback: callback,
+        identity: identity,
+        baseUrl: baseUrl,
+        cookiePath: cookiePath,
+        refreshRetryPeriod: 12345,
+        useCookie: false,
+      });
+      uid2.init({
+        callback: jest.fn(() => {
+          return 'testing one';
+        }),
+      });
+      uid2.init({
+        callback: jest.fn(() => {
+          return 'testing two';
+        }),
+      });
+    });
+    test('should contain only one init callback function', () => {
+      const initCallbacks = uid2.getInitCallbacks();
```

Comment:
It's better to test the outcome we want - in this case, we could set a new identity and make sure all the callback functions were called.

Reply:
I've changed this!